pkgs/applications/graphics/loupe/default.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv
+, lib
+, fetchgit
+, rustPlatform
+, meson
+, pkg-config
+, cmake
+, gtk4
+, itstool
+, desktop-file-utils
+, ninja
+, gnome-builder
+, wrapGAppsHook4
+, libadwaita
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "loupe";
+  version = "44.alpha";
+
+  src = fetchgit {
+    url = "https://gitlab.gnome.org/BrainBlasted/loupe.git";
+    rev = "aa781d386485d76a647a7b8968b91a08e521c780";
+    sha256 = "sha256-pkHmwnK9lBjpFQbGkEjgTi5rOsYLMK6OydB/k6aMk84=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    hash = "sha256-ay2HNTcYumkwpUZkntJK4vA7puXz5swb8NwC8LG8vW8=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    pkg-config
+    cmake
+    gtk4
+    itstool
+    desktop-file-utils # update-desktop-database
+    ninja
+    gnome-builder
+    wrapGAppsHook4
+  ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
+
+  buildInputs = [
```

Comment:
Not sure how Loupe loads images but if it relies on `gdk-pixbuf` internally, you might want to look on `eog` for how to deal with `webp` support.

Reply:
It does use gdk-pixbuf but it doesn't seem to support gif and webp yet so I think I can save that for another time

https://gitlab.gnome.org/BrainBlasted/loupe/-/issues/27

---

My main outstanding question is about `./result/share/dbus-1/services/org.gnome.Loupe.service` & `loupe --gapplication-service`

Reply:
That sounds like the issue is just about animation. Most webp images should be just static: https://developers.google.com/speed/webp/gallery1

Reply:
`ERROR loupe::widgets::image_page > Could not load image: Unrecognised image file format`

I guess I could try the `eog` bits and see if it fixes things

Reply:
I've added the eog parts and I was able to open a webp file

IDK if I needed all of these parts so I've put it in a separate WIP commit.
I might look at trimming it down depending on what is actually necessary